rs/replicated_state/src/canister_state/system_state/call_context_manager.rs

```diff
@@ -768,12 +776,17 @@ impl CallContextManager {
 
     /// Registers a callback for an outgoing call.
     pub fn register_callback(&mut self, callback: Callback) -> CallbackId {
-        self.stats.on_register_callback(&callback);
-
         self.next_callback_id += 1;
         let callback_id = CallbackId::from(self.next_callback_id);
-        self.callbacks.insert(callback_id, Arc::new(callback));
 
+        self.stats.on_register_callback(&callback);
+        if callback.deadline != NO_DEADLINE {
+            assert!(self
+                .unexpired_callbacks
+                .insert((callback.deadline, callback_id)));
+        }
+
+        self.callbacks.insert(callback_id, Arc::new(callback));
```

Comment:
The old code didn't assert for callbacks not being already present at all. Now we do assert for the unexpired callbacks whereas we don't for the callbacks themselves. Is there a reason making the check for `unexpected_callbacks` stricter than the other one?

Reply:
I believe that I had asserts for both and then removed the one for `self.callbacks` precisely because it wasn't there before. But apparently didn't think much about what it meant to hold on to the other.

It's probably safer to remove both. (To the extent that we won't crash and burn anyway, since we've just inserted two callbacks with the same callback ID.)